GoogleForCreators/web-stories-wp
# Pre-Publish Checklist: Update Thumbnails for PagePreview Cache

## Context

Last release, prometheus introduced a cool new performance win with the `PagePreview` used in the carousel not realizing that the checklist is using that same component to render pages in the checklist. The initial viewing of the checklist if there's page previews happening caches as a black rectangle and doesn't update until you collapse the checklist once. 

## Summary

Conditionally rendering the checklist's page preview thumbnails so that what they only render when needed and don't cache early before they're accurate.

## Relevant Technical Choices

Moving the `openPanel` state out of the base `checklist` and into the provider that's used to track the checklist's current status so that we know what panel is open in the checklist and can pass `isVisible` where needed from the different sets of checks.
Still overriding the `PagePreview` button to be a div so that it's not double nesting buttons


## To-do



## User-facing changes

- If the thumbnail caching experiment is on and you open the checklist with one or more of the 4 checks that will render a thumbnail visible you will no longer get a black rectangle the first time you open the checklist (see videos). 

**Before (Main)**
https://user-images.githubusercontent.com/10720454/134074641-eb5393a4-20b8-4fbd-8124-03e8e8c3b1ae.mp4

**After (This branch)**
https://user-images.githubusercontent.com/10720454/134074645-35ac9650-ed49-48bb-9f41-e07ff8bb70f2.mp4



## Testing Instructions

The four checks you want to look at are: 
- Too little text
- Too much text
- Too many links 
- Bad color contrast between bg and fg 

Just make sure that the page on the thumbnail's actually visible and when you update the page the thumbnail also updates.




This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #8883
